api/inc/uvisor_exports.h

```diff
@@ -78,10 +78,15 @@
  *
  * This macro works from both inside and outside function scope.
  *
- * FIXME This is currently not implemented. This issue is tracked at
- * https://github.com/ARMmbed/uvisor/issues/288
+ * The implementations differ due to compilation differences, c++ static_assert
+ * is known from c++11 while mbed-os compiles with c++98
  */
-#define UVISOR_STATIC_ASSERT(cond, msg)
+#ifdef  __cplusplus
```

Comment:
We should use a different ifdef here.

Consider checking for C11 (`__STDC_VERSION`) first, and then falling back to other implementations.